bdd_tests/features/steps/database.py

```diff
@@ -0,0 +1,252 @@
+    assert context.connection is not None, "connection should be established"
+
+
+@when(u"I close database connection")
+def disconnect_from_database(context):
+    """Close the connection to database."""
+    context.connection.close()
+    context.connection = None
+
+
+@then(u"I should be disconnected")
+def check_disconnection(context):
+    """Check that the connection has been closed."""
+    assert context.connection is None, "connection should be closed"
+
+
+@given(u"CCX Notification database is created for user {user} with password {password}")
+@when(u"CCX Notification database is created for user {user} with password {password}")
+def database_is_created(context, user, password):
+    """Perform connection to CCX Notification database to check its ability."""
+    connect_to_database(context, "notification", user, password)
+
+
+@given(u"CXX Notification Writer database contains all required tables")
+def database_contains_all_tables(context):
+    """Check if CCX Notification database contains all required tables."""
+    raise NotImplementedError(u'STEP: Given CXX Notification Writer database contains all tables')
+
+
+@given(u"CCX Notification Service database is set up")
+def ensure_database_is_set_up(context):
+    """Check that the tables exist in the DB."""
+    # at least following tables should exist
+    tables = ("migration_info",
+              "new_reports",
+              "notification_types",
+              "reported",
+              "states")
+
+    cursor = context.connection.cursor()
+    for table in tables:
+        try:
+            cursor.execute("SELECT 1 from {}".format(table))
+            v = cursor.fetchone()
+            context.connection.commit()
+        except UndefinedTable as e:
+            context.connection.rollback()
+            raise e
+    pass
+
+
+@given(u"CCX Notification database is empty")
+def notification_db_empty(context):
+    """Ensure that the CCX Notification database has no reports, but has all tables."""
+    # We actually only want `new_reports` and `reported` table to be empty, so let's clean up
+    cursor = context.connection.cursor()
+    try:
+        cursor.execute("TRUNCATE TABLE new_reports")
+        cursor.execute("TRUNCATE TABLE reported")
+        context.connection.commit()
+    except Exception as e:
+        context.connection.rollback()
+        raise e
+
+
+@when(u"I select all rows from table {table}")
+def select_all_rows_from_table(context, table):
+    """Select number of all rows from given table."""
+    cursor = context.connection.cursor()
+    try:
+        cursor.execute("SELECT count(*) as cnt from {}".format(table))
+        results = cursor.fetchone()
+        assert len(results) == 1, "Wrong number of records returned: {}".format(len(results))
+        context.query_count = results[0]
+    except Exception as e:
+        raise e
+
+
+@then(u"I should get {expected_count:d} row")
+@then(u"I should get {expected_count:d} rows")
```

Comment:
maybe we can use regex https://behave.readthedocs.io/en/stable/regular_expressions.html (I'm not sure and it's not important)

Reply:
It's too much work =)